tls/extensions/s2n_server_supported_versions.c

```diff
@@ -64,22 +78,29 @@ int s2n_extensions_server_supported_versions_process(struct s2n_connection *conn
     return 0;
 }
 
-int s2n_extensions_server_supported_versions_recv(struct s2n_connection *conn, struct s2n_stuffer *extension)
+static int s2n_server_supported_versions_recv(struct s2n_connection *conn, struct s2n_stuffer *in)
 {
-    S2N_ERROR_IF(s2n_extensions_server_supported_versions_process(conn, extension) < 0, S2N_ERR_BAD_MESSAGE);
+    if (!s2n_is_tls13_enabled()) {
+        return S2N_SUCCESS;
+    }
 
-    return 0;
+    S2N_ERROR_IF(s2n_extensions_server_supported_versions_process(conn, in) < 0, S2N_ERR_BAD_MESSAGE);
+    return S2N_SUCCESS;
 }
 
-int s2n_extensions_server_supported_versions_send(struct s2n_connection *conn, struct s2n_stuffer *out)
-{
-    int extension_length = s2n_extensions_server_supported_versions_size(conn);
+/* Old-style extension functions -- remove after extensions refactor is complete */
 
-    GUARD(s2n_stuffer_write_uint16(out, TLS_EXTENSION_SUPPORTED_VERSIONS));
-    GUARD(s2n_stuffer_write_uint16(out, extension_length - 4));
+int s2n_extensions_server_supported_versions_size(struct s2n_connection *conn)
+{
+    return 6;
```

Comment:
/nit Replace 6 with a constant value

Reply:
I'm not sure that's necessary here. The constant would only be used in one place, and this method will be deleted soon anyway after the full refactor. I'm trying to avoid rewriting doomed code.